src/shared/parameter_v2/scripts/c_config.py

```diff
@@ -0,0 +1,121 @@
+from c_parameter import CParameter
+from typing import List, Set
+
+#######################################################################
+#                              C Config                               #
+#######################################################################
+
+
+class CConfig(object):
+
+    DEFINITION = "typedef struct {name}_s {{{contents}}} {name}_t;\n"
+    FORWARD_DECLERATION = "typedef struct {name}_s {name}_t;\n"
+    MALLOC = "{name}_t* {name}_config = ({name}_t*)malloc(sizeof({name}_t));\n"
+    INITIALIZATION = "{name}_t {name}_init = {{{contents}}};\n"
+    INCLUDED_CONFIG_INITIALIZATION = ".{name} = {name}_config,\n"
+    MEMCPY = "memcpy({name}_config, &{name}_init, sizeof({name}_t));\n"
+    FREE = "free((void*){ptr_to_instance});\n"
+    INCLUDE_CONFIG = "const {name}_t* {name};\n"
+
+    def __init__(self, config_name: str, ptr_to_instance: str):
+        """Initializes a CConfig with the given name. The corresponding
+        generation strings (definition, forward_decleration, malloc, initialization,
+        memcpy, free, include_config) are available from read-only properties.
+
+        :param config_name: The name of the config (i.e the filename of the yaml)
+        :param ptr_to_instance: A string representation of where this config
```

Comment:
There is, but in python3 we can leave it out because its explicitly enforced in the arguments, if you try calling this function with absurd types it will complain